### PR TITLE
Flav/qdrant create collection script

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

See https://github.com/dust-tt/dust/issues/5175 and [specifications](https://www.notion.so/dust-tt/Design-Doc-Qdrant-re-arch-d0ebdd6ae8244ff593cdf10f08988c27).

This PR introduces a script to create a collection on the future cluster. The script performs the following steps:
1. Creates the collection with:
   - `shard_method`: "custom"
   - `shard_number`: 2
   - `replication_factor`: 2

2. Creates the shards on the collection with:
   - `shard_key`: "key_{0..23}"
   - `shards_number`: 2
   - `replication_factor`: 2
   - `placement`: [] // During collection creation, the cluster assigns shards to peers.

Additionally, this PR updates the `docker-compose.yml` to include two Qdrant nodes. This is necessary because enabling [distributed mode](https://qdrant.tech/documentation/guides/distributed_deployment/#enabling-distributed-mode-in-self-hosted-qdrant) is required to create shard keys. And distributed mode needs at least two nodes/peers.

⚠️ Note: All Qdrant data will be lost locally.

## Risk

Works locally.

## Deploy Plan

To create the collection on a cluster:
```
➜   cargo run --bin qdrant_create_collection -- --cluster <cluster> --provider <provider_id> --model <model_id>
```

### CODE REVIEW

File: core/src/data_sources/qdrant.rs [hunk 1]
```diff
@@ -555,4 +580,8 @@ impl DustQdrantClient {
             }
         }
     }
+
+    pub fn raw_client(&self) -> Arc<QdrantClient> {
```
Comment (flvndvd) [inline core/src/data_sources/qdrant.rs:584]: @spolu, since https://github.com/dust-tt/dust/pull/5148, I haven't been able to access the Qdrant client. I didn't want to expose all those methods in our client, so I created this method to gain access to the raw client.

File: core/bin/qdrant/create_collection.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                collection_name.clone(),
+                &qdrant::shard_key::Key::Keyword(format!("{}_{}", client.shard_key_prefix(), i)),
+                Some(2),
+                Some(2),
+                &[],
+            )
+            .await?;
+    }
+
+    match res.result {
```
Comment (spolu) [inline core/bin/qdrant/create_collection.rs:117]: Should be before the shard creations

File: core/bin/qdrant/create_collection.rs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        })
+        .await?;
+
+    // Then, we create the 24 shard_keys.
+    for i in 0..SHARD_KEY_COUNT {
+        raw_client
+            .create_shard_key(
+                collection_name.clone(),
+                &qdrant::shard_key::Key::Keyword(format!("{}_{}", client.shard_key_prefix(), i)),
+                Some(2),
```
Comment (spolu) [inline core/bin/qdrant/create_collection.rs:110]: Big question is why do we need to repeat these configs at shard creation.

Mind asking them on slack?

**Review by spolu — APPROVED** [review]
LGTM

Left a few nits

File: core/bin/qdrant/create_collection.rs (hunk 2 above)
Comment (flvndvd) [reply core/bin/qdrant/create_collection.rs:104]: Updated.

File: core/bin/qdrant/create_collection.rs (hunk 3 above)
Comment (flvndvd) [reply core/bin/qdrant/create_collection.rs]: Asked [here](https://dust4ai.slack.com/archives/C05HBLG1TRB/p1716449483486159).

TL;DR: They are not required when creating a shard key.
